src/Layer/Layer.svelte

```diff
@@ -0,0 +1,45 @@
+<script>
+  import { getContext, setContext } from "svelte";
+
+  /**
+   * Specify the layer level and override any existing levels based on hierarchy.
+   * @type {0 | 1 | 2 }
+   */
+  export let level = undefined;
+
+  /**
+   * Specify the HTML element to render. If none is specified, a `div` is rendered.
+   * @type {typeof import("svelte").SvelteComponent}
+   */
+  export let as = "div";
+
+  /**
+   * Specify the Layer HTML element props
+   * @type {import('svelte/elements').HTMLElementAttributes}
+   */
+  export let layerProps = {};
+
+  /** Set an id for the Layer component */
+  export let id = "ccs-" + Math.random().toString(36);
+
+  let ref = null;
+
+  // If there is no level override, determine the Level based on the hierarchy
+  const parentLevel = getContext("LAYER_CONTEXT");
+  if (level === undefined) {
+    level = typeof parentLevel === "number" ? parentLevel + 1 : 0;
+  }
+  setContext("LAYER_CONTEXT", level);
```

Comment:
- This does not account for dynamically changing levels, which would require the use of a store in the context and a reactive statement to update the local level/store on change. Not sure if this is worth supporting, though.
- Personally, I would prefix context keys consistently to preempt collisions, e.g. use something like `'carbon.layer'`/`'carbon-components.layer'` in this case. If something like this is done, it should apply everywhere, though. So maybe a separate PR would make more sense.

Reply:
Can we make the context reactive? Could something like this work?

```svelte
  $: parentLevel = getContext("LAYER_CONTEXT");
  $: {if (level === undefined) {
      level = typeof parentLevel === "number" ? parentLevel + 1 : 0;
    }
    setContext("LAYER_CONTEXT", level);
  }
```

Will take advice on the approach to the context name.

Reply:
This will not work, contexts are fixed, they are set during child component construction.
That is why you would need a store.

Reply:
Alright. I'll wait to see what others say about whether or not to use stores.